### PR TITLE
73 user profile and setting

### PR DESCRIPTION
**Pull request by cloud9ight**

### CODE REVIEW

File: front-end/src/views/DashBoardMainContent/DashBoardMainContentView.tsx
Comment (thogb) [inline front-end/src/views/DashBoardMainContent/DashBoardMainContentView.tsx:1]: Need to revert this. I already changed. see this https://github.com/TomatoesBurner/ArtFusion-AI-Hub/pull/76/files

Comment (cloud9ight) [reply front-end/src/views/DashBoardMainContent/DashBoardMainContentView.tsx:1]: (dashboard)/dashboard/page.tsx is the old version of the user dashboard page. we don't need the side bar right now. 
For the FeaturesSection and GallerySection functionality still need more time

Comment (thogb) [reply front-end/src/views/DashBoardMainContent/DashBoardMainContentView.tsx:1]: I mean revert back the change for this.
image
1 should be in /views
2 and 3 should be kept as it is in /components.

Then 1 should be called in /app/(dashboard)/dashboard/page.tsx
like 
...
<DashboardView/>
...

I have already made the changes for these.
image

File: front-end/src/components/DashBoardLayout/UserPopover/UserPopover.tsx [hunk 1]
```diff
@@ -74,6 +95,9 @@ const UserPopover = () => {
           <ListItemText>Logout</ListItemText>
         </MenuItem>
       </Menu>
+
+      {showProfile && <Profile onClose={handleCloseProfile} />}
```
Comment (thogb) [inline front-end/src/components/DashBoardLayout/UserPopover/UserPopover.tsx:99]: This component I made is a popover like click and dropdown menu with three options. 
The above two should not be called here. They should be two new pages with corresponding /views and page.tsx created like. /app(dashboard)/settings/profile/page.tsx. Clicking on the option will navigate to the page, like how github right top is doing.

File: front-end/src/api/userApi.ts [hunk 2]
```diff
@@ -6,4 +6,10 @@ export class UserApi {
   public static async getMe() {
     return (await appApi.get("/users/me")).data as ApiResponseDto<UserMeDto>;
   }
+
+  // update user info
+  public static async updateUser(data: Partial<UserMeDto>) {
```
Comment (thogb) [inline front-end/src/api/userApi.ts:11]: Haven't made one for this yet, will try to make one this week.

File: front-end/src/components/DashBoardLayout/UserPopover/Profile.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            <TextField
+              fullWidth
+              label="Name"
+              name="name"
+              value={formData.name}
+              onChange={handleChange}
+              variant="outlined"
+            />
+          </Grid>
+          <Grid item xs={12}>
```
Comment (thogb) [inline front-end/src/components/DashBoardLayout/UserPopover/Profile.tsx:76]: Don't think it is option for user to change email, should be disabled. Maybe let them update first last name, and theme.

File: front-end/src/components/DashBoardLayout/UserPopover/Settings.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+}
+
+const Settings: React.FC<{ onClose: () => void }> = ({ onClose }) => {
+  const dispatch = useDispatch();
+  const user = useSelector((state: { user: User }) => state.user);
+
+  const [formData, setFormData] = useState<User>({
+    name: user.name || "",
+    email: user.email || "",
+    notificationsEnabled: user.notificationsEnabled || false,
```
Comment (thogb) [inline front-end/src/components/DashBoardLayout/UserPopover/Settings.tsx:33]: We have no notification feature.

File: front-end/src/components/DashBoardLayout/UserPopover/Settings.tsx [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+    language: user.language || "English",
+    twoFactorEnabled: user.twoFactorEnabled || false, // Two-Factor Authentication state
```
Comment (thogb) [inline front-end/src/components/DashBoardLayout/UserPopover/Settings.tsx:35]: Won't give then option to disable this.

File: front-end/src/components/DashBoardLayout/UserPopover/Settings.tsx [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    setFormData((prevData) => ({
+      ...prevData,
+      [name]: type === "checkbox" ? checked : value,
+    }));
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    try {
+      await UserApi.updateUser(formData);
```
Comment (thogb) [inline front-end/src/components/DashBoardLayout/UserPopover/Settings.tsx:59]: Two updateUser page don't make much sense

**Comment by cloud9ight** [comment]
@thogb Ready to merge, thanks

**Comment by cloud9ight** [comment]
[Settings and user profile](https://trello.com/c/dZzDcwKi/56-settings-and-user-profile)

**Pull request merged**